metagraph/src/annotation/row_diff_builder.cpp

```diff
@@ -428,14 +418,14 @@ void build_pred_succ(const graph::DeBruijnGraph &graph,
                   outfbase);
 
     std::optional<sdsl::bit_vector> dummy;
-    auto* succinct = dynamic_cast<graph::DBGSuccinct const*>(&graph);
-    if (succinct) {
+    if (auto* succinct = dynamic_cast<graph::DBGSuccinct const*>(&graph)) {
         dummy = succinct->get_boss().mark_all_dummy_edges(num_threads);
```

Comment:
Wouldn't it be better to do `succinct->mask_dummy_kmers(num_threads, false)` here or somewhere on the way to this call? Then we can get rid of annoying dummy checks below.

Reply:
Well, here goes nothing. I've tried to add it in some way. I guess, should be further discussed in the relevant section of the review.